test/distributed/org/apache/cassandra/distributed/test/FqlReplayDDLExclusionTest.java

```diff
@@ -0,0 +1,104 @@
+            try (com.datastax.driver.core.Cluster c = com.datastax.driver.core.Cluster.builder().addContactPoint("127.0.0.1").build();
+                 Session s = c.connect())
+            {
+                s.execute("CREATE KEYSPACE fql_ks WITH replication = {'class': 'SimpleStrategy', 'replication_factor': 1};");
+
+                node.nodetool("enablefullquerylog", "--path", temporaryFolder.getRoot().getAbsolutePath());
+
+                s.execute("CREATE TABLE fql_ks.fql_table (id int primary key);");
+                s.execute("INSERT INTO fql_ks.fql_table (id) VALUES (1)");
+
+                node.nodetool("disablefullquerylog");
+            }
+
+            // here we are dropping and we expect that ddl replay will reconstruct it
+
+            node.executeInternal("DROP TABLE fql_ks.fql_table;");
+
+            final ToolRunner.Runners runners = new ToolRunner.Runners();
+
+            // without --replay-ddl-statements, the replay will fail on insert because underlying table is not there
+            final ToolRunner negativeRunner = runners.invokeClassAsTool("org.apache.cassandra.fqltool.FullQueryLogTool",
+                                                                        "replay",
+                                                                        "--keyspace", "fql_ks",
+                                                                        "--target", "127.0.0.1",
+                                                                        "--", temporaryFolder.getRoot().getAbsolutePath());
+
+            assertEquals(0, negativeRunner.getExitCode());
+
+            try
+            {
+                node.executeInternalWithResult("SELECT * from fql_ks.fql_table");
+                fail("This query should fail because we do not expect fql_ks.fql_table to be created!");
+            }
+            catch (final Exception ex)
+            {
+                assertTrue(ex.getMessage().contains("table fql_table does not exist"));
+            }
+
+            // here we replay with --replay-ddl-statements so table will be created and insert will succeed
+            final ToolRunner positiveRunner = runners.invokeClassAsTool("org.apache.cassandra.fqltool.FullQueryLogTool",
+                                                                        "replay",
+                                                                        "--keyspace", "fql_ks",
+                                                                        "--target", "127.0.0.1",
+                                                                        // important
+                                                                        "--replay-ddl-statements",
+                                                                        "--", temporaryFolder.getRoot().getAbsolutePath());
+
+            assertEquals(0, positiveRunner.getExitCode());
+            assertTrue(node.executeInternalWithResult("SELECT * from fql_ks.fql_table;").hasNext());
```

Comment:
can you also assert the rows?

Reply:
row utils will handle it for you, so just need to show that only a single row with the value of 1 is present.

```
assertRows(node.executeWithResult("SELECT * from fql_ks.fql_table"),
                       QueryResults.builder().row(1).build());
```

that will test that only one row exists and the only value is `1`